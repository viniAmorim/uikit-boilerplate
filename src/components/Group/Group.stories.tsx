import React from 'react';

import { GroupProps } from '@mantine/core';
import { Story, Meta } from '@storybook/react';

import Button from '../Button';
import Group from './Group';

export default {
  title: 'Atoms/Display/Group',
  component: Group,
} as Meta;

const Template: Story<GroupProps> = (args) => <Group {...args} />;

export const Default = Template.bind({});
Default.args = {
  children: (
    <>
      <Button variant="secondary">Button 1</Button>
      <Button variant="secondary">Button 2</Button>
      <Button variant="secondary">Button 3</Button>
    </>
  ),
};

export const Spacing = Template.bind({});
Spacing.args = {
  spacing: 'xl',
  children: (
    <>
      <Button variant="secondary">Button 1</Button>
      <Button variant="secondary">Button 2</Button>
      <Button variant="secondary">Button 3</Button>
    </>
  ),
};

export const Grow = Template.bind({});
Grow.args = {
  grow: true,
  children: (
    <>
      <Button variant="secondary">Button 1</Button>
      <Button variant="secondary">Button 2</Button>
      <Button variant="secondary">Button 3</Button>
    </>
  ),
};
