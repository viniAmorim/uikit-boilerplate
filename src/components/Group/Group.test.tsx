import React from 'react';

import { render } from '@testing-library/react';

import Group from './Group';

describe('Group', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Group>
        <div>1</div>
        <div>2</div>
      </Group>,
    );

    const group = container.firstChild;

    expect(group).toBeInTheDocument();
    expect(group).toMatchSnapshot();
    expect(group).toHaveStyle(`
    display: flex;
    flex-direction: row;
  `);
  });
  it('should render children with correct flexbox properties', () => {
    const { container } = render(
      <Group position="center" align="center" noWrap>
        <div>1</div>
        <div>2</div>
      </Group>,
    );
    const group = container.firstChild;
    expect(group).toHaveStyle(`
      justify-content: center;
      align-items: center;
      flex-wrap: nowrap;
    `);
  });
});
