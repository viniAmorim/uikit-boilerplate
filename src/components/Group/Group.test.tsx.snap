// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Group should render correctly 1`] = `
<div
  class="mantine-Group-root mantine-k3ov3c"
>
  <div>
    1
  </div>
  <div>
    2
  </div>
</div>
`;
